src/core/Akka.Streams/Dsl/Graph.cs

```diff
@@ -424,6 +424,194 @@ public MergePreferred(int secondaryPorts, bool eagerClose = false)
+                        }
+                        else
+                        {
+                            buffer.Enqueue(inlet);
+                        }
+                    }, onUpstreamFinish: () =>
+                    {
+                        if (_stage.EagerComplete)
+                        {
+                            _stage.In.ForEach(Cancel);
+                            runningUpstreams = 0;
+                            if (!HasPending) CompleteStage();
+                        }
+                        else
+                        {
+                            runningUpstreams -= 1;
+                            if (UpstreamsClosed && !HasPending) CompleteStage();
+                        }
+                    });
+                }
+
+                SetHandler(_stage.Out, this);
+            }
+
+            public override void PreStart()
+            {
+                foreach (var input in _stage.In)
+                {
+                    TryPull(input);
+                }
+            }
+
+            public override void OnPull()
+            {
+                if (!HasPending)
+                    DequeueAndDispatch();
+            }
+
+            public bool HasPending => allBuffers.Any(c => c.NonEmpty);
+
+            public bool UpstreamsClosed => runningUpstreams == 0;
+
+            private void DequeueAndDispatch()
+            {
+                var input = SelectNextElement();
+                Push(_stage.Out, Grab(input));
+                if (UpstreamsClosed && !HasPending)
+                    CompleteStage();
+                else
+                    TryPull(input);
+            }
+
+            private Inlet<T> SelectNextElement()
+            {
+                var tp = 0;
+                var ix = 0;
+
+                while (ix < _stage.In.Count)
+                {
+                    if (allBuffers[ix].NonEmpty)
+                    {
+                        tp += _stage.Priorities[ix];
+                    }
+                    ix += 1;
+                }
+
+                int r = randomGen.Next(tp);
+                Inlet<T> next = null;
+                ix = 0;
+
+                while (ix < _stage.In.Count && next == null)
+                {
+                    if (allBuffers[ix].NonEmpty)
+                    {
+                        r -= _stage.Priorities[ix];
+                        if (r < 0)
+                            next = allBuffers[ix].Dequeue();
+                    }
+                    ix += 1;
+                }
+
+                return next;
+            }
+
+            public override string ToString() => "MergePrioritized";
+        }
+        #endregion
+
+        /// <summary>
+        /// Create a new <see cref="T:Akka.Streams.Dsl.MergePrioritized`1" /> with specified number of input ports.
+        /// </summary>
+        /// <param name="priorities">Priorities of the input ports</param>
+        public MergePrioritized(IList<int> priorities)
```

Comment:
Should I change it to `IEnumerable<int>`? Originally it was `Seq[Int]`

Reply:
I would say yes, no need for a list since no add/remove operations are needed.